hw/net/rocker/rocker_world.c

```diff
@@ -97,10 +108,8 @@ enum rocker_world_type world_type(World *world)
 
 const char *world_name(World *world)
 {
-    switch (world->type) {
-    case ROCKER_WORLD_TYPE_OF_DPA:
-        return "OF_DPA";
-    default:
-        return "unknown";
+    if (world) {
```

Comment:
Agree. I really wanted as assert.. but don't see it being used anywhere.
Also wanted to somewhat mimic the default: case in the earlier code.

Removing the check.

On Sun, Apr 19, 2015 at 9:19 PM, Scott Feldman <notifications@github.com>
wrote:

> In hw/net/rocker/rocker_world.c
> <https://github.com/pbhide/qemu-rocker/pull/1#discussion_r28663372>:
>
> > @@ -97,10 +108,8 @@ enum rocker_world_type world_type(World *world)
> >
> >  const char *world_name(World *world)
> >  {
> > -    switch (world->type) {
> > -    case ROCKER_WORLD_TYPE_OF_DPA:
> > -        return "OF_DPA";
> > -    default:
> > -        return "unknown";
> > +    if (world) {
>
> Shouldn't need to check if world is non-null here. If NULL was passed in,
> that's a bug in the calling code and I think we'd rather have the device
> crash (so the bug can get fixed) rather than returning something that sort
> of look like world is non-NULL.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pbhide/qemu-rocker/pull/1/files#r28663372>.
>